tools/Azure.Mcp.Tools.Monitor/src/Commands/WebTests/WebTestsCreateOrUpdateCommand.cs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.CommandLine.Parsing;
+using Azure.Mcp.Core.Commands;
+using Azure.Mcp.Core.Models.Option;
+using Azure.Mcp.Tools.Monitor.Models.WebTests;
+using Azure.Mcp.Tools.Monitor.Options;
+using Azure.Mcp.Tools.Monitor.Options.WebTests;
+using Azure.Mcp.Tools.Monitor.Services;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.Mcp.Tools.Monitor.Commands.WebTests;
+
+public sealed class WebTestsCreateOrUpdateCommand(ILogger<WebTestsCreateOrUpdateCommand> logger) : BaseMonitorWebTestsCommand<WebTestsCreateOrUpdateOptions>
+{
+    private const string _commandTitle = "Create or updates a web test in Azure Monitor";
+    private const string _commandName = "createorupdate";
+
+    private readonly ILogger<WebTestsCreateOrUpdateCommand> _logger = logger;
+
+    private readonly Option<string> _resourceNameOption = MonitorOptionDefinitions.WebTest.WebTestResourceName;
+    private readonly Option<string> _appInsightsComponentIdOption = MonitorOptionDefinitions.WebTest.AppInsightsComponentId;
+    private readonly Option<string> _resourceLocationOption = MonitorOptionDefinitions.WebTest.ResourceLocation;
+    private readonly Option<string> _locationsOption = MonitorOptionDefinitions.WebTest.Locations;
+    private readonly Option<string> _requestUrlOption = MonitorOptionDefinitions.WebTest.RequestUrl;
+
+    private readonly Option<string> _webTestNameOption = MonitorOptionDefinitions.WebTest.WebTestName;
+    private readonly Option<string> _descriptionOption = MonitorOptionDefinitions.WebTest.Description;
+    private readonly Option<bool> _enabledOption = MonitorOptionDefinitions.WebTest.Enabled;
+    private readonly Option<int> _expectedStatusCodeOption = MonitorOptionDefinitions.WebTest.ExpectedStatusCode;
+    private readonly Option<bool> _followRedirectsOption = MonitorOptionDefinitions.WebTest.FollowRedirects;
+    private readonly Option<int> _frequencyInSecondsOption = MonitorOptionDefinitions.WebTest.FrequencyInSeconds;
+    private readonly Option<string> _headersOption = MonitorOptionDefinitions.WebTest.Headers;
+    private readonly Option<string> _httpVerbOption = MonitorOptionDefinitions.WebTest.HttpVerb;
+    private readonly Option<bool> _ignoreStatusCodeOption = MonitorOptionDefinitions.WebTest.IgnoreStatusCode;
+    private readonly Option<bool> _parseRequestsOption = MonitorOptionDefinitions.WebTest.ParseRequests;
+    private readonly Option<string> _requestBodyOption = MonitorOptionDefinitions.WebTest.RequestBody;
+    private readonly Option<bool> _retryEnabledOption = MonitorOptionDefinitions.WebTest.RetryEnabled;
+    private readonly Option<bool> _sslCheckOption = MonitorOptionDefinitions.WebTest.SslCheck;
+    private readonly Option<int> _sslLifetimeCheckInDaysOption = MonitorOptionDefinitions.WebTest.SslLifetimeCheckInDays;
+    private readonly Option<int> _timeoutInSecondsOption = MonitorOptionDefinitions.WebTest.TimeoutInSeconds;
```

Comment:
Newer design choice is using the RHS values directly where the LHS variables are used